src/jsonsettings.h

```diff
@@ -0,0 +1,27 @@
+#include <QIODevice>
+#include <QJsonParseError>
+#include <QJsonObject>
+#include <QSettings>
+
+// Use for QSetting to read json
+static bool readJSONSettings(QIODevice &device, QMap<QString, QVariant> &map) {
+  QJsonParseError error;
+  QJsonObject json = QJsonDocument::fromJson(device.readAll(), &error).object();
+  if(error.error != QJsonParseError::NoError)
+      return false;
+  const auto keys = json.keys();
+  for (const auto &key : keys)
+    map.insert(key, json.value(key).toVariant());
+  return true;
+}
+// Use for QSettings to write JSON
+static bool writeJSONSettings(QIODevice &device, const QMap<QString, QVariant> &map) {
+    QJsonObject temp;
+    const auto keys = map.keys();
+    for(const auto &key : keys)
+        temp.insert(key, map.value(key).toJsonValue());
```

Comment:
```suggestion
    QJsonObject temp = QJsonObject::fromVariantMap(map);
```

I think that's equivalent.

Reply:
+1